### DISCUSSION TITLE
Word-wrap in album view

### DISCUSSION THREAD

**Discussion by kamidon** [jcorporation/myMPD#1030] [Ideas]
I recently switched to using mpd + myMPD for playback on my main music listening system. It's wonderful but does have one issue I haven't find a way around yet. In the album view the album names are truncated and I often can't distinguish among multiple albums. If the album title wrapped to I second line I think I would be able to distinguish them. Is there some way that could be made available optionally? Alternatively, adding an option to view the albums in a table similar to the filesystem view could also work.

This may be a problem that is mostly unique to me as I listen to a lot of "classical" (western concert) music and I have most of my collection tagged so that individual pieces, which often have multiple movements and thus tracks, are "albums" with the album name set to the composer and title of the work. I can't tell the albums apart by the cover art because pieces that come from the same recording (e.g. ripped CD or downloaded digital recording) usually have the same cover art for all the pieces. With my setup, the truncated album title usually just shows the composer and the first word or so of the title, which isn't enough to distinguish different works with the same composer.

Thanks for myMPD and for considering my issue!

**Comment by jcorporation** [comment]
The complete title displays on mouse over. 

You know that mpd supports the composer tag and myMPD can display this tag also in the album view?

**Reply by kamidon** [reply]
I did find the mouseover but feel it doesn't really solve my use case for three reasons:
1. I typically use myMPD from a tablet where there is no mouse pointer to hover over items
2. On my primary desktop computer I use a Wacom tablet as a pointer (legacy of RSI issues) and while it isn't impossible to hover with it, it is difficult because small movements of the pen tip, which are hard to avoid, dismiss the mouseover.
3. On my laptop, where I typically use a trackpad, hover works okay but is time-consuming to find the right entry.

Thank you for the reminder about the composer tag. My current tagging scheme is a compromise that I created 5-6 years ago that attempted to ensure all the metadata I cared about was available in all the different music programs I typically used at that time. I'm sure things have evolved and at some point I should definitely re-evaluate whether there is sufficiently broad support for additional tags like composer that I can reduce some of the "packing" of information into the core fields. However, retagging everything would be a *major* project even scripting the process to help speed it up, so I'd like to avoid that in the near-term if possible.

I definitely understand if you feel these suggestions are not appropriate for your vision for myMPD. I appreciate the consideration of my needs. I'm not much of a web developer (I mainly do more backend and embedded development) but if either of the ideas I suggested sound like they could be incorporated and you can give me some recommendations about how to approach implementation I'd give it a try.

**Reply by jcorporation** [reply]
I quickly looked into this. Commit 31defc6bf339fa7403c4fc511eee86a70f35a64f enables line wrapping in footer of cards in the database view.

**Reply by kamidon** [reply]
Thank you for the quick response! I haven't yet rebuilt my local copy of mympd with this patch (I probably won't get to that until the weekend) but with the patch as a guide I was able to find the right place to tweak the CSS in the Firefox web inspector view so I could see what it would look like. It definitely solves my problem and works well in a regular browser (and presumably on my tablet as well). I'll try it out on my Android phone when I build with the patch.  Here is a screenshot:

![image](https://github.com/jcorporation/myMPD/assets/7191814/540d0e76-f7d7-429e-bdb0-672df3807173)

**Comment by kamidon** [comment]
Thanks very much for merging the text wrapping of the labels in album view. Before I realized you'd already released a new version with that and another fix I built the prior version with this patch and it is exactly what I was hoping for. I'll pickup the full version when my upstream distribution updates.

**Comment by GK-774** [comment]
If one would not want text wrapping of the labels in album view (I don´t, I prefer the more compact view we had before), the only way is to revert that commit, correct?

**Reply by jcorporation** [reply]
Correct, but I can add an option to toggle this behavior.

**Reply by kamidon** [reply]
For what it is worth, it makes sense to me that this behavior is something that could be preferred or not depending on how people of chosen to manage the tags on their music collection. I'm in favor of making a toggle for it. Hopefully that is straightforward and doesn't require a lot of work.

**Comment by GK-774** [comment]
Yes, an option would be nice, preferably as a browser specific setting (on a tablet I would prefer more compact "no wrap", while on a large screen it might be "wrap"). Ideally the "elements per page" setting could be changed to a variable instead of a fixed list, again as browser specific setting. This way one could dial in an ideal setting for every device with the album view filling one page.

**Reply by jcorporation** [reply]
What should be the default value of the word wrap option?

**Reply by kamidon** [reply]
Even though I'll be overriding word wrap to "on" (wrapping happens) on all my clients, I'd suggest "off" (no wrap) might be the better default since it leads to more predictable layouts, especially on smaller-screen devices.
